astroquery/cadc/cadctap/core.py

```diff
@@ -0,0 +1,411 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+=============
+TAP plus
+=============
+
+"""
+from astroquery.utils.tap.core import TapPlus
+from astroquery.cadc.cadctap.tapconn import TapConnCadc
+from astroquery.cadc.cadctap.jobSaxParser import JobSaxParserCadc
+import requests
+import os.path
+from astropy.extern.six.moves.urllib.parse import urlencode
+
+try:
+    # python 3
+    import http.client as httplib
+except ImportError:
+    # python 2
+    import httplib
+
+__all__ = ['TapPlusCadc']
+
+VERSION = "1.0.1"
+TAP_CLIENT_ID = "aqtappy-" + VERSION
+DEFAULT_LOGIN_URL = 'http://www.canfar.phys.uvic.ca/ac/login'
+DEFAULT_COOKIE_PREFIX = 'CADC_SSO='
+
+
+class TapPlusCadc(TapPlus):
+    """TAP plus class
+    Provides TAP and TAP+ capabilities
+    Reason for change
+    -----------------
+    To override some parts of the utils/tap library in order for it to
+    function with the CADC's implementation of TAP
+    """
+
+    def __init__(self, url=None, host=None, server_context=None,
+                 tap_context=None, port=80, sslport=443,
+                 default_protocol_is_https=False, connhandler=None,
+                 verbose=True):
+        """Constructor
+        Parameters
+        ----------
+        url : str, mandatory if no host is specified, default None
+            TAP URL
+        host : str, optional, default None
+            host name
+        server_context : str, optional, default None
+            server context
+        tap_context : str, optional, default None
+            tap context
+        port : int, optional, default '80'
+            HTTP port
+        sslport : int, optional, default '443'
+            HTTPS port
+        default_protocol_is_https : bool, optional, default False
+            Specifies whether the default protocol to be used is HTTPS
+        connhandler connection handler object, optional, default None
+            HTTP(s) connection hander (creator). If no handler is provided, a
+            new one is created.
+        verbose : bool, optional, default 'True'
+            flag to display information about the process
+        Reason for change

```

Comment:
add an empty line before this and change the heading to be something numpydoc compatible, e.g: `Notes`